improver_tests/utilities/temporal/test_TimezoneExtraction.py

```diff
@@ -170,22 +171,44 @@ def test_create_output_cube(with_cell_method):
 @pytest.mark.parametrize("include_time_coord", (True, False))
 def test_check_input_cube_dims(include_time_coord):
     """Checks that check_input_cube_dims can differentiate between an input cube
-    with time, y, x coords and one where time is missing. Also checks that timezone_cube
-    has been reordered correctly."""
+    with time, y, x coords and one where time is missing."""
     cube = make_input_cube([3, 4])
     timezone_cube = make_timezone_cube()
     plugin = TimezoneExtraction()
+    plugin.timezone_cube = timezone_cube
     if include_time_coord:
-        plugin.check_input_cube_dims(cube, timezone_cube)
-        assert plugin.timezone_cube.coord_dims("UTC_offset") == tuple(
-            [plugin.timezone_cube.ndim - 1]
-        )
+        plugin.check_input_cube_dims(cube)
+        assert cube.coord_dims("time") == tuple([cube.ndim - 1])
     else:
         cube.remove_coord("time")
         with pytest.raises(
             ValueError, match=r"Expected coords on input_cube: time, y, x "
         ):
-            plugin.check_input_cube_dims(cube, timezone_cube)
+            plugin.check_input_cube_dims(cube)
+
+
+def test_check_aux_time_coord():
+    """Checks that check_input_cube_dims can work with a auxiliary time
+    coordinate. This occurs when partial periods are allowed. In these
+    cases the lower time bound may be the same between different time
+    points, preventing the creation of a dimension coordinate that
+    required monotonically increasing bounds. In this case a anonymous
+    coordinate is created by iris that can be named and used for
+    reordering.
+
+    A temporary name should be assigned to the time dimension for
+    reordering. This should not be present on the output, but the time
+    coordinate should have been moved to the last dimension."""
+    cube = make_input_cube([3, 4])
+    timezone_cube = make_timezone_cube()
+    iris.util.demote_dim_coord_to_aux_coord(cube, "time")
+
+    plugin = TimezoneExtraction()
+    plugin.timezone_cube = timezone_cube
+    plugin.check_input_cube_dims(cube)
+    print(cube)
+    assert cube.coord_dims("time") == tuple([cube.ndim - 1])
+    assert "time_points" not in [crd.name() for crd in cube.coords()]
```

Comment:
Checking for a negative...

What we expect here is that we have exactly one dimension coordinate that is anonymous. How about
```
assert len(cube.dim_coords) == len(cube.shape) - 1
```
I'd rather check that the coordinate associated with the auxilliary `time` coord is the one that has no dimension coord, but I haven't figured that out yet.

Reply:
I've added your suggested test to highlight that an anonymous dimension exists on the output. I've retained the explicit check that the "time_points" coord has been removed from the output as well just to demonstrate that this temporary coordinate won't escape the context of this function and head downstream.